setup.py

```diff
@@ -142,6 +142,10 @@ def copy_data_tree(datadir, destdir):
 gdal_major_version = int(gdal_version_parts[0])
 gdal_minor_version = int(gdal_version_parts[1])
 
+if gdal_major_version == 1 and gdal_minor_version < 11:
+    log.fatal("GDAL >= 1.11 is required for rasterio.  Please upgrade GDAL.")
```

Comment:
I wonder if we shouldn't also be detecting when GDAL changes underneath Rasterio, given that different GDAL versions require different Rasterio shims?  That would require stashing the compile-time version of GDAL somewhere, but I think `setuptools` has machinery for generic metadata coupled with `pkg_resources` for fetching at runtime.  I know we could at least drop the GDAL version into a text file and include it in `pkg_data`.  I have no idea if that would be as slow as entrypoints (#711).

Dynamically creating an additional Python file somewhere at install time and populating it with the GDAL string is an option that avoids I/O aside from an additional input.  Something much cleaner than:

```
with open('rasterio/compile_version.py', 'w') as f:
    f.write("""
compile_version = '{}'
""".format(subprocess.check_output(['gdalinfo', '--version'])
```

Reply:
@geowurster : this adds a good bit of complexity, so if this is something we'd like to do, I'd prefer we handle it in a dedicated PR.  I agree that it would be nice to offer up a useful error message if we detect that the environment has changed on a subsequent run, rather than failing with a very obscure and hard to troubleshoot exception from stepping into code that applies to the wrong version of GDAL.

Is some of this mitigated by the rasterio wheels that ship with a minimal GDAL?

Reply:
I'm not aware of any Python project that does this. I don't think we need to be more rigorous about this than Numpy, lxml, or GDAL's own Python bindings. At least not yet.

Reply:
Agreed.  Mostly just thinking out loud.